api/v1beta1/machinedeployment_types.go

```diff
@@ -25,6 +25,10 @@ const (
 	// MachineDeploymentTopologyFinalizer is the finalizer used by the topology MachineDeployment controller to
 	// clean up referenced template resources if necessary when a MachineDeployment is being deleted.
 	MachineDeploymentTopologyFinalizer = "machinedeployment.topology.cluster.x-k8s.io"
+
+	// MachineDeploymentFinalizer is the finalizer used by the MachineDeployment controller to
+	// cleanup the MachineDeployment descendant MachineSets when a MachineDeployment is being deleted.
+	MachineDeploymentFinalizer = "machinedeployment.cluster.x-k8s.io"
```

Comment:
I believe that would be appropriate while looking into the [docs](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#finalizers:~:text=Identifiers%20of%20custom%20finalizers%20consist%20of%20a%20domain%20name%2C%20a%20forward%20slash%20and%20the%20name%20of%20the%20finalizer) and [validation](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/util/validation/validation.go#L42).

I don't think we ever agreed on a pattern in the issue linked, but if we are adding this now we should make sure to cement a pattern

Reply:
Yup agree. Now would be the time to figure out the correct pattern. `apiGroup/<lower-case kind>` for the finalizer of our "main controller" reconciling that object seems like something that would work (note: the controller name of our "main" controller is also lower-case kind)

(I'm saying "main" controller because in a few cases we have two controllers and the additional ones are named like this: "topology/machineset", "topology/machinedeployment", "topology/cluster")


@fabriziopandini @vincepri @chrischdi @enxebre opinions?

Reply:
the pattern suggested above makes sense to me

Reply:
@vincepri WDYT (given your recent discussion in Slack: https://kubernetes.slack.com/archives/C0EG7JC6T/p1724853254310249?thread_ts=1721923881.095799&cid=C0EG7JC6T)